test/client/enketo-webform.spec.js

```diff
@@ -0,0 +1,1731 @@
+import i18next from 'i18next';
+import lodash from 'lodash';
+import applicationCache from '../../public/js/src/module/application-cache';
+import connection from '../../public/js/src/module/connection';
+import controller from '../../public/js/src/module/controller-webform';
+import events from '../../public/js/src/module/event';
+import formCache from '../../public/js/src/module/form-cache';
+import gui from '../../public/js/src/module/gui';
+import settings from '../../public/js/src/module/settings';
+import store from '../../public/js/src/module/store';
+
+/**
+ * @typedef {import('sinon').SinonStub<Args, Return>} Stub
+ * @template Args
+ * @template Return
+ */
+
+/**
+ * @typedef {import('sinon').SinonSandbox} Sandbox
+ */
+
+/**
+ * @typedef {import('../../app/models/survey-model').SurveyObject} Survey
+ */
+
+/** @type {Record<string, any> | null} */
+let webformPrivate = null;
+
+describe( 'Enketo webform app', () => {
+    /** @type {string} */
+    let enketoId;
+
+    /** @type {Record<string, string>} */
+    let defaults;
+
+    /** @type {Sandbox} */
+    let sandbox;
+
+    /** @type {import('sinon').SinonFakeTimers} */
+    let timers;
+
+    /** @type {HTMLElement} */
+    let mainElement = null;
+
+    /** @type {HTMLElement} */
+    let loaderElement = null;
+
+    before( async () => {
+        const formHeader = document.querySelector( '.form-header' );
+
+        if ( formHeader == null ) {
+            const domParser = new DOMParser();
+            const formDOM = domParser.parseFromString( `
+                <div class="main">
+                    <div class="paper">
+                        <div class="form-header"></div>
+                    </div>
+                </div>
+                <div class="main-loader"></div>
+            `, 'text/html' );
+
+            mainElement = formDOM.documentElement.querySelector( '.main' );
+            loaderElement = formDOM.documentElement.querySelector( '.main-loader' );
+
+            document.body.append( mainElement, loaderElement );
```

Comment:
These are accessed by enketo-webform at the top level on import/load, to define variables used by other functions during initialization. While those could also be wrapped in a function to provide for more deliberate behavior in test, this had to be done first before I felt changes like that would be safe. I'm up to change that now if preferred.